### ISSUE TITLE
cmake configuration error: compiling a test executable failed

### ISSUE THREAD

**Issue by bishesh** [ethz-asl/libnabo#67]
I'm trying to install from source in ubuntu 16.04 but configuring using cmake fails with the following error:
```
OpenCL disabled, not looking for it

 ANN not found, disabling it in benchmarks

 FLANN not found, disabling it in benchmarks

 CMake Error at tests/CMakeLists.txt:95 (message):
   Compiling a test executable failed.
 Call Stack (most recent call first):
   tests/CMakeLists.txt:104 (try_compile_cloud_types)
```
Attached are the cmakecache and log files (log file renamed since uploading .log is not supported).
[CMakeCache.txt](https://github.com/ethz-asl/libnabo/files/1196734/CMakeCache.txt)

[CMakeOutputlog.txt](https://github.com/ethz-asl/libnabo/files/1196747/CMakeOutputlog.txt)

**Comment by SaiPrabhaMergu** [comment]
I face the same issue, How did you solve this? , could you share the solution

**Comment by HannesSommer** [comment]
Which compiler are you using? It is tested with (mostly vanilla) 16.04. And I cannot reproduce myself neither with g++ 5.4 nor clang 3.8.

Could you please provide the output of `cmake --trace ` ?

**Comment by bishesh** [comment]
@HannesSommer The output of cmake --trace is attached here. My g++ version: g++ (Ubuntu 5.4.0-6ubuntu1~16.04.4) 5.4.0 20160609.
@SaiPrabhaMergu The problem is not solved yet.

[cmakeTrace.txt](https://github.com/ethz-asl/libnabo/files/1358988/cmakeTrace.txt)

**Comment by HannesSommer** [comment]
Very strange. Given these versions I must assume the difference is somewhere else. Maybe in `cmake`? (could you please report `cmake --version`). 
Also, based on that idea could you please try out the PR #71  (branch `fix/removeOutOfPlaceCommas`), and report back if it fixes the issue?

**Comment by bishesh** [comment]
ccmake version 3.5.1 
I also tried PR #71 removing commas, but I still get exactly the same error.

**Comment by SaiPrabhaMergu** [comment]
I am trying to build it on windows, using cmake. The error which I see is as follows:

> CMake Error at tests/CMakeLists.txt:95 (message):
>   Compiling a test executable failed.
> Call Stack (most recent call first):
>   tests/CMakeLists.txt:104 (try_compile_cloud_types)

I have tried  PR #71, but still it failed.

**Comment by HannesSommer** [comment]
Thanks for the feedback. This seems to be a tough one. As I cannot reproduce I need more help from you. Could you please try #72 and upload the output of `cmake --trace 2>&1` (assuming `bash` syntax to merge the error stream into the standard output) ?

**Comment by SaiPrabhaMergu** [comment]
I am trying to build this code for Visual Studio 2015 x64.
cmake-gui version :3.3
Eigen version:3.3.4
Boost version:1.65.1
grep version:2.5.4
I face another error now. 

> cmake warning Policy CMP0054 is not set:

Can I get your comment on this

**Comment by HannesSommer** [comment]
I have no experience with libnabo on Windows. Maybe others do?

In any case: you posed a warning (that should not be triggered through the changes in #72). Are you sure that it is your actual problem?

**Comment by bishesh** [comment]
Thanks @HannesSommer ! Trying #72 helped me figure out the problem. In the error output I could now see "Eigen/Core not found". The EIGEN_INCLUD_DIR path I gave was <top-most-eigen-dir>/Eigen, but I now see that I should give it <top-most-eigen-dir>. It should have issued proper error of not getting the expected path for Eigen! Anyway, thanks for your help. I have now successfully built the library.

**Comment by HannesSommer** [comment]
Awesome. This is great news! This issue helped improving the feedback of the CMake code!

**Comment by SaiPrabhaMergu** [comment]
I was able to build the library too, Following are the steps I followed
1) To escape error: "cmake warning Policy CMP0054 is not set:" add line "cmake_policy(SET CMP0054 OLD)" in CMakeLists.txt(I added in line 2)
2) I also had to remove the line  "add_definitions(-Wall -Wextra)" line 44 in CMakeLists.txt, this was stopping the nabo.lib from forming.
The above steps helped me solve the issues. So I thought I should share.
